pyvista/core/filters/data_set.py

```diff
@@ -4340,6 +4340,52 @@ def shrink(self, shrink_factor=1.0, progress_bar=False):
             return output.extract_surface()
         return output
 
+    def tessellate(self, progress_bar=False):
+        """Tessellate a mesh.
+
+        This filter approximates nonlinear FEM-like elements with linear
+        simplices. The output mesh will have geometry and any fields specified
+        as attributes in the input mesh's point data. The attribute's copy
+        flags are honored, except for normals.
+        
+        For more details see `vtkTessellatorFilter <https://vtk.org/doc/nightly/html/classvtkTessellatorFilter.html#details>`_.
+
+        progress_bar : bool, optional
+            Display a progress bar to indicate progress.
+
+        Parameters
+        ----------
+        progress_bar : bool, optional
+            Display a progress bar to indicate progress.
+
+        Returns
+        -------
+        pyvista.DataSet
+            Dataset with tessellated mesh.  Return type matches input.
+
+        Examples
+        --------
+        First, plot the original cube.
+
+        >>> import pyvista
+        >>> mesh = pyvista.Cube()
+        >>> mesh.plot(show_edges=True, line_width=5)
+
+        Now, plot the tessellated mesh.
+
+        >>> tessellated = mesh.tessellate()
+        >>> tessellated.clear_data()  # cleans up plot
+        >>> tessellated.plot(show_edges=True, line_width=5)
+
+        """
+        alg = _vtk.vtkTessellatorFilter()
+        alg.SetInputData(self)
+        _update_alg(alg, progress_bar, 'Tessellating Mesh')
+        output = pyvista.wrap(alg.GetOutput())
+        if isinstance(self, _vtk.vtkPolyData):
+            return output.extract_surface()
```

Comment:
When using PolyData the following warning occurred. I would make the change that this method is not supported by PolyData.
```bash
2022-02-10 23:18:38.696 ( 130.113s) [        D0348C00]vtkTessellatorFilter.cx:1243  WARN| vtkTessellatorFilter (0x3a5ad20): Input dataset has cells without parameterizations (VTK_POLYGON,VTK_POLY_LINE,VTK
_POLY_VERTEX,VTK_TRIANGLE_STRIP,VTK_CONVEX_POINT_SET,VTK_QUADRATIC_POLYGON). They will be ignored. Use vtkTriangleFilter, vtkTetrahedralize, etc. to parameterize them first.
```
```suggestion
        if isinstance(self, _vtk.vtkPolyData):
            raise TypeError('Tessellate filter is not support PolyData object.')
```